test.js

```diff
@@ -1,11 +1,180 @@
-/* eslint-disable no-new */
 'use strict';
-var fn = require('./');
+const test = require('ava');
+const path = require('path');
+const concatStream = require('concat-stream');
+const fork = require('child_process').fork;
 
-fn();
+function streamToPromise(stream) {
+	return new Promise((resolve, reject) => {
+		stream.on('error', reject);
+		stream.pipe(concatStream({encoding: 'string'}, resolve));
+	});
+}
+
+// setTimeout as a Promise. Use it to delay via `await`
+function tick(time) {
+	time = time || 0;
+	if (process.env.TRAVIS) {
+		// slow things down for reliable tests on Travis-CI
+		time *= 10;
+	}
+	return new Promise(resolve => setTimeout(resolve, time || 0));
+}
+
+test.beforeEach(t => {
+	const child = fork(path.join(__dirname, 'fixture.js'), {silent: true});
+
+	const exit = new Promise((resolve, reject) =>
+			child.on('exit', code =>
+					(code > 0 ? reject : resolve)(code)
+			)
+	);
+
+	t.context = {
+		// Tell the child to create a promise, and reject it
+		rejectWithError: (key, message) => child.send({action: 'reject-error', key, message}),
+		rejectWithValue: (key, value) => child.send({action: 'reject-value', key, value}),
+		rejectWithNothing: key => child.send({action: 'reject-nothing', key}),
+
+		// Tell the child to handle the promise previously rejected
+		handle: key => child.send({action: 'handle', key}),
+
+		// Tell the child to reinstall loudRejection
+		reinstall: () => child.send({action: 'reinstall'}),
 
-new Promise(function (resolve, reject) {
-	setImmediate(function () {
-		reject(new Error('unicorn'));
+		// Kill the child (returns a promise for when the child is done).
+		kill: () => {
+			child.kill();
+			return exit;
+		},
+
+		// The stdout of the child. Useful for debug.
+		stdout: streamToPromise(child.stdout),
+
+		// The stderr of the child. This is where unhandledRejections will be logged.
+		stderr: streamToPromise(child.stderr),
+
+		// Promise for when the child has exited
+		exit
+	};
+
+	child.on('message', message => {
+		if (message.status !== 'ready') {
+			t.fail(`I got a message I do not understand: ${JSON.stringify(message)}`);
```

Comment:
Ahh.
That's an old habit to avoid escape characters. Going to have to unlearn that one.